sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_cluster.asciidoc

```diff
@@ -0,0 +1,200 @@
+= sycl_ext_oneapi_cluster_group
+
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2024-2025 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 8 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+Nvidia's compute capability (cc) 9.0 devices introduced a new level in the thread hierarchy, called as thread block clusters, in CUDA terminology. A thread block cluster, is a collection of thread blocks (a workgroup in SYCL terminology), which are guaranteed to be scheduled to be scheduled on a separate compute unit, and are guaranteed to run in parallel, and the thread blocks in the cluster can access each other's shared memory (local memory SYCL terminology). This has various applications, convolutions, GEMMs and FFTs to name a few.
```

Comment:
We typically keep to an 80 character line limit to make it easier for cmd line editors and limiting the scope of future changes.

Reply:
No worries, thanks :)